rawhttp-core/src/main/java/rawhttp/core/RawHttp.java

```diff
@@ -404,4 +404,11 @@ private RequestLine verifyHost(RequestLine requestLine, RawHttpHeaders.Builder h
         }
     }
 
+    private String composeHostPort(URI uri) {
```

Comment:
This method can be static... also, it seems to be used solely to set the value of the host header, so I suggest changing its name to `hostHeaderValueFor(URI uri)`.